src/net/rpc.rs

```diff
@@ -363,11 +515,25 @@ impl ConversationHttp {
         }
     }
 
+    fn handle_load_stacks_chain_tip<W: Write>(http: &mut StacksHttp, fd: &mut W, req: &HttpRequestType, burndb: &mut BurnDB) -> Result<Option<(BurnchainHeaderHash, BlockHeaderHash)>, net_error> {
+        match burndb.get_canonical_chain_tip_headers()? {
```

Comment:
I don't think this is the right method.  The `get_canonical_chain_tip_headers()` method doesn't consider whether or not the Stacks block at the tip even _exists_, let alone is present on-disk and validated.

I encountered this problem in my mempool-relayer code.  I ended up solving it by adding a new method to `StacksChainState` that would return the canonical chain tip based on which anchored blocks it already has processed.  Happy to add that to the downloader branch so you can use it.

Reply:
Should a node be responding to these requests if it doesn’t have data for its canonical chain tip?

Reply:
It _can't_ respond to these endpoints if it tries to open a MARF trie that doesn't exist, so at the very least, you will need to verify that the Stacks block has been processed by the chainstate first.

But to answer your question, the canonical Stacks chain as determined by just the block-commit transactions is _not_ the canonical Stacks chain as determined by actually processing the blocks.  For example, the block-commit with the most sortitions might correspond to a non-existent block, or an invalid block.

Reply:
Yep -- that makes sense, the node should be able to continue processing requests even if it doesn't have data for the canonical chain tip. My preference would be to leave this as is for now, and then update this line when the mempool-relayer PR opens.

Reply:
Can you add a `TODO` in the code and open an issue to track this?

Reply:
Added #1388

Reply:
Thanks!